file-formats/dsfi/type/zif_aff_dsfi_v1.intf.abap

```diff
@@ -28,6 +33,15 @@ INTERFACE zif_aff_dsfi_v1
       "! <p class="shorttext">Engine</p>
       "! Engine
       engine               TYPE ty_engine,
+      "! <p class="shorttext">AMDP Reference</p>
+      "! The name of the AMDP Implementation Reference
+      "! $required
+      amdp_name            TYPE c LENGTH 100,
+      "! <p class="shorttext">Automatically exposed in SQL Service</p>
+      "! Flag whether or not the SQL impementation of the
+      "! CDS SCalar Function will be automatically exposed in all SQL Services
+      "! $required
+      auto_exposure_flag   TYPE abap_bool,
```

Comment:
Regarding your question about increasing the format version: Are both fields really mandatory in the future? They are marked as `$required`. If so, I think we must increase the format version. If both fields are non-mandatory fields the change would be compatible.

See also: https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#format-versions-and-compatibility

Reply:
Hi @schneidermic0 , I understand this would generally require an increase in the format version. However, there is no consumption of the AFF for DSFI yet in abapGit outside of SAP. I am in the process of implementing/delivering both the zabapGit and abapGit-Steampunk handlers for this object, so we expect no customers to be affected by this change.

The standard IF_AFF_DSFI_V1 has already been released to customers with the fields flagged as required without a version increase. Another important point is that required flag is set for UI purposes. The objects can be created even if these fields are empty.

I had separate discussions with @wurzka and my PO today. Katharina offered some suggestions of how we could make the change compatible (making the fields not required anymore or packing them into a non-required structure), but the product would prefer not to change the standard IF_AFF_DSFI_V1 if not strictly needed.

Alternatively, I could imagine having the fields not be required in ZIF_AFF_DSFI_V1 and keeping them flagged as required in IF_AFF_DSFI_V1. This should not be a problem, since the fields are not strictly speaking mandatory.

Another option would be to increase the version on the Z interface but not on the standard one.

The obvious problem with these two alternatives is that then we have a split between the Z and standard interfaces. My opinion is that this would be a bigger issue, as it might create confusion in the future.

I'd also like to question whether the Z interface is needed at all. The object is AFF native, so the standard interface will surely exist in any system which supports it. Should/could we discontinue ZIF_AFF_DSFI_V1?

Reply:
> However, there is no consumption of the AFF for DSFI yet in abapGit outside of SAP.

If this is the case, you can think of changing the file format without increasing the format version. But you need to keep in mind, that you might have to update your real implementation in existing on-premise versions. This is what would be used by the abapGit object handlers.

> Alternatively, I could imagine having the fields not be required in ZIF_AFF_DSFI_V1 and keeping them flagged as required in IF_AFF_DSFI_V1. This should not be a problem, since the fields are not strictly speaking mandatory.

This is not an option from my point of view. Basically, the `ZIF_AFF_DSFI_V1` and `IF_AFF_DSFI_V1` shall be identical. This repository shall provide the information how the ABAP file format for DSFI is specified.
We have chosen to add a Z in front of the interface name so that externals can suggest/contribute changes from their systems.

> I had separate discussions with @wurzka and my PO today. Katharina offered some suggestions of how we could make the change compatible (making the fields not required anymore or packing them into a non-required structure), but the product would prefer not to change the standard IF_AFF_DSFI_V1 if not strictly needed.

I had also a short chat with @wurzka. I understood she suggested to move all fields required for engine = "SQL" to a non-required structure, like `sql_properties` which contains the mandatory fields. This, actually, make sense to me. And it would keep even the format version compatible. ;)
If you need more details on this, feel also free to schedule a call with @wurzka, (maybe, your PO?) and me.

Reply:
There's still a compatibility issue. I suppose that's the added enum value sqlEngine to the engine field, which did not have a default.
That should be ok, no? Given that before there was only one enum value, it was as good as a default. And the new AFF now has analyticalEngine as the default.

Reply:
> I suppose that's the added enum value sqlEngine to the engine field, which did not have a default.

As far as I can see, there is a default value defined, isn't it?
https://github.com/SAP/abap-file-formats/blob/942689d56cf21d5a9afbc39dd42d546200118be4/file-formats/dsfi/type/zif_aff_dsfi_v1.intf.abap#L9

I think the automatic check complains about your addition of `sql_properties` which is, however, fine from my point of view.
@albertmink is my understanding correct?